services/federation/ap.go

```diff
@@ -0,0 +1,24 @@
+package federation
+
+import "github.com/yitsushi/go-misskey/core"
+
+// ShowRequest is the request used by the show command.
+// Contains a single URI.
+type ShowRequest struct {
+	URI string `json:"uri"`
+}
+
+// ShowResponse is an empty response for the Show command.
+type ShowResponse struct {
+}
+
+// Show shows the ActivityPub object specified by the URI.
+func (s *Service) Show(request ShowRequest) error {
```

Comment:
I never expose the request to the caller as we can make changes on it like local validations later, or updating given fields based on the input.

For single value, I just create a single argument.
https://github.com/yitsushi/go-misskey/blob/main/services/antennas/delete.go#L16

If there are more than one (or more than two, depending on the endpoint) parameters, I just create an Options struct, even if it's 1:1 match.
https://github.com/yitsushi/go-misskey/pull/29/files#diff-a931b78a2fb7862a2893f4ace437ce1d05404401bb081eefcd550d86e23d8765R36-R38

Reply:
As an addition, I'm thinking simply just not exporting request structs and all comments go for Options and/or the function itself.